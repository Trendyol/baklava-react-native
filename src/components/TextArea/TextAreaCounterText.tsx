import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import { getCounterText, getTextColor } from './utils';

export const TextAreaCounterText = React.memo(
  ({
    counterText,
    errorState,
    maxLength,
  }: {
    counterText?: number;
    errorState: boolean;
    maxLength: number;
  }): JSX.Element | null => {
    const textColor = getTextColor({ errorState });

    const content = getCounterText({
      counterText,
      errorState,
    });

    return (
      <Box mt="3xs" ml="m">
        <Text variant="subtitle04Regular" color={textColor}>
          {`${content}/${maxLength}`}
        </Text>
      </Box>
    );
  },
);
