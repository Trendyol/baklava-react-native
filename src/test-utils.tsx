import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
