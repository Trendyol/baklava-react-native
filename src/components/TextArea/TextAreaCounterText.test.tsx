import React from 'react';
import { render } from '../../test-utils';
import { TextAreaCounterText } from './TextAreaCounterText';

describe('TextArea Counter Text', () => {
  let counterText: number;
  let errorState: boolean;
  const DEFAULT_MAX_LENGTH = 200;

  test('should render counter text correctly', () => {
    // given
    counterText = 0;
    errorState = false;

    // when
    const { toJSON } = render(
      <TextAreaCounterText
        counterText={counterText}
        errorState={errorState}
        maxLength={DEFAULT_MAX_LENGTH}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render null when not has error text', () => {
    // given
    counterText = 0;
    errorState = false;

    // when
    const { toJSON } = render(
      <TextAreaCounterText
        counterText={counterText}
        errorState={errorState}
        maxLength={DEFAULT_MAX_LENGTH}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render with custom maxLength', () => {
    // given
    counterText = 0;
    errorState = false;
    const customMaxLength = 500;

    // when
    const { toJSON } = render(
      <TextAreaCounterText
        counterText={counterText}
        errorState={errorState}
        maxLength={customMaxLength}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
