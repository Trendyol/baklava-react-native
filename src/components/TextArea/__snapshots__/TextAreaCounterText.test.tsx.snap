// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Counter Text should render counter text correctly 1`] = `
<View
  accessible={false}
  style={
    [
      {
        "marginLeft": 16,
        "marginTop": 4,
      },
    ]
  }
>
  <Text
    accessible={false}
    style={
      [
        {
          "color": "#95A1B5",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 14,
          "textAlign": "left",
        },
      ]
    }
  >
    0/200
  </Text>
</View>
`;

exports[`TextArea Counter Text should render null when not has error text 1`] = `
<View
  accessible={false}
  style={
    [
      {
        "marginLeft": 16,
        "marginTop": 4,
      },
    ]
  }
>
  <Text
    accessible={false}
    style={
      [
        {
          "color": "#95A1B5",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 14,
          "textAlign": "left",
        },
      ]
    }
  >
    0/200
  </Text>
</View>
`;

exports[`TextArea Counter Text should render with custom maxLength 1`] = `
<View
  accessible={false}
  style={
    [
      {
        "marginLeft": 16,
        "marginTop": 4,
      },
    ]
  }
>
  <Text
    accessible={false}
    style={
      [
        {
          "color": "#95A1B5",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 14,
          "textAlign": "left",
        },
      ]
    }
  >
    0/500
  </Text>
</View>
`;
